sklearn/decomposition/tests/test_nmf.py

```diff
@@ -159,8 +178,42 @@ def test_sparse_transform():
     A[A > 1.0] = 0
     A = csc_matrix(A)
 
-    model = nmf.NMF()
-    A_fit_tr = model.fit_transform(A)
-    A_tr = model.transform(A)
-    # This solver seems pretty inconsistent
-    assert_array_almost_equal(A_fit_tr, A_tr, decimal=2)
+    for solver in ('proj-grad', 'coordinate'):
+        model = nmf.NMF(solver=solver, random_state=1, tol=1e-4)
+        A_fit_tr = model.fit_transform(A)
+        A_tr = model.transform(A)
+        assert_array_almost_equal(A_fit_tr, A_tr, decimal=2)
+
+
+@ignore_warnings
+def test_non_negative_matrix_factorization_path():
+    # Test path consistency between the class and the public function
```

Comment:
This is to test that `non_negative_matrix_factorization` public function is called in the same way, either directly or through the `NMF` class.